### PR TITLE
Add docs templates, guidelines

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/.feature-template-uno-only.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+ * Detail how to add the feature to an app and get it working. Provide code snippets.
+ * Provide specific examples addressing common use cases
+
+## Supported features by platform
+
+<!-- If not feature is supported on every platform, fill in the matrix below to show which ones are supported. If there's no limitations to the feature on any platform, consider removing this whole section. -->
+
+| Feature        |  Windows  | Android |  iOS  |  Web (WASM)  | macOS | Linux (Skia)  | Win 7 (Skia) | 
+|---------------|-------|-------|-------|-------|-------|-------|
```
Comment (Xiaoy312) [inline doc/.feature-template-uno-only.md:22]: missing a column here to make the table work:
```suggestion
|---------------|-------|-------|-------|-------|-------|-------|-|
```

File: doc/.feature-template-uno-only.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+<!-- For available Markdown syntax, check out https://docs.microsoft.com/contribute/contribute/how-to-write-use-markdown -->
```
Comment (Xiaoy312) [inline doc/.feature-template-uno-only.md:2]: The link 404'd for me. Are you able to access it, and in an incognito session too?
Maybe use the github one?: https://guides.github.com/features/mastering-markdown/

File: doc/.feature-template.md [hunk 3]
```diff
@@ -1,68 +1,30 @@
...
 
-<!-- Use below format to display note
-> [!NOTE]
-Some note
+<!-- Leave the infotip below in place, and add a link to the UWP documentation for the feature or control you're documenting. If the feature has no UWP equivalent, you should be using the Uno-only feature template: .feature-template-uno-only.md -->
 
-> [!IMPORTANT]
-Some important note
+> [!TIP]
+> This article covers Uno-specific information for YourFeature. For a full description of the feature and instructions on using it, consult the UWP documentation: https://docs.microsoft.com/en-us/uwp/api/yourfeature
```
Comment (Xiaoy312) [inline doc/.feature-template.md:9]: uppercase the "need to change" part, so we don't miss it?:
```suggestion
> This article covers Uno-specific information for YourFeature. For a full description of the feature and instructions on using it, consult the UWP documentation: https://docs.microsoft.com/en-us/uwp/api/INSERT_UWP_FEATURE_HERE
```

File: doc/.feature-template.md [hunk 4]
```diff
@@ -1,68 +1,30 @@
...
-Some warning note
--->
+ * Add a short intro here
 
-## Syntax
+## Supported features
 
-```csharp
+| Feature        |  Windows  | Android |  iOS  |  Web (WASM)  | macOS | Linux (Skia)  | Win 7 (Skia) | 
+|---------------|-------|-------|-------|-------|-------|-------|
```
Comment (Xiaoy312) [inline doc/.feature-template.md:16]: same, missing a column here to make the table work:
```suggestion
|---------------|-------|-------|-------|-------|-------|-------|-|
```

**Review by Xiaoy312 — APPROVED** [review]

File: doc/.feature-template-uno-only.md (hunk 2 above)
Comment (ajpinedam) [reply doc/.feature-template-uno-only.md:2]: Same for me. The link is returning 404

Comment (davidjohnoliver) [reply doc/.feature-template-uno-only.md:2]: Thanks, I'll update to the github link - I copied it from the existing file without testing it.

Comment (davidjohnoliver) [inline doc/.feature-template-uno-only.md:2]:
```suggestion
<!-- For available Markdown syntax, check out https://guides.github.com/features/mastering-markdown/ -->
```

File: doc/.feature-template.md [hunk 5]
```diff
@@ -1,68 +1,30 @@
...
-description: describe your feature implementation
-keywords: uno platform, your feature
-dev_langs:
-  - csharp
----
 
-<!-- To know about all the available Markdown syntax, Check out https://docs.microsoft.com/contribute/contribute/how-to-write-use-markdown -->
-<!-- Ensure you remove all comments before submission, to ensure that there are no formatting issues when displaying this page.  -->
-<!-- It is recommended to check how the Documentation will look in the sample app, before Merging a PR -->
+<!-- For available Markdown syntax, check out https://docs.microsoft.com/contribute/contribute/how-to-write-use-markdown -->
```
Comment (davidjohnoliver) [inline doc/.feature-template.md:2]:
```suggestion
<!-- For available Markdown syntax, check out https://guides.github.com/features/mastering-markdown/ -->
```

File: doc/.howto-template.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# How to [complete TaskX]
+
+<!-- For available Markdown syntax, check out https://docs.microsoft.com/contribute/contribute/how-to-write-use-markdown -->
```
Comment (davidjohnoliver) [inline doc/.howto-template.md:3]:
```suggestion
<!-- For available Markdown syntax, check out https://guides.github.com/features/mastering-markdown/ -->
```

**Review by ajpinedam — APPROVED** [review]

**Pull request merged**